File: algorithm1/binary_search/python/search_insert_position.py
import math

class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        pivot = 0
        left = 0
        right = len(nums) -1
        
        while(left <= right):
          pivot = math.floor((left + right) / 2)
          
          if nums[pivot] == target:
            return pivot
          elif nums[pivot] > target:
            right = pivot - 1
          else:
            left = pivot + 1
          
        return left
            

File: algorithm1/binary_search/SearchInsertPosition.ts
function searchInsert(nums: number[], target: number): number {
    let pivot: number;
    let left = 0, right = nums.length - 1;

    while (left <= right) {
      pivot = Math.floor((left + right) / 2);

      if (nums[pivot] === target) {
        return pivot;
      } else if (nums[pivot] > target) {
        right = pivot - 1;
      } else {
        left = pivot + 1;
      }
    }
  
  return left;
};